analysis/study_definition.py

```diff
@@ -1,33 +1,191 @@
-from cohortextractor import StudyDefinition, patients, codelist, codelist_from_csv
+from cohortextractor import StudyDefinition, patients, codelist, codelist_from_csv, filter_codes_by_category
 
-chronic_cardiac_disease_codes = codelist_from_csv(
-    "codelists/opensafely-chronic-cardiac-disease.csv", system="ctv3", column="CTV3ID"
-)
-chronic_liver_disease_codes = codelist_from_csv(
-    "codelists/opensafely-chronic-liver-disease.csv", system="ctv3", column="CTV3ID"
-)
-salbutamol_codes = codelist_from_csv(
-    "codelists/opensafely-asthma-inhaler-salbutamol-medication.csv",
-    system="snomed",
-    column="id",
-)
-systolic_blood_pressure_codes = codelist(["2469."], system="ctv3")
-diastolic_blood_pressure_codes = codelist(["246A."], system="ctv3")
+from codelists import *
+
+def first_diagnosis_in_period(dx_codelist):
+    return patients.with_these_clinical_events(
+        dx_codelist,
+        returning="date",
+        find_first_match_in_period=True,
+        include_month=True,
+        return_expectations={
+            "incidence": 0.2,
+            "date": {"earliest": "1950-01-01", "latest": "today"},
+        },
+    )
+
+def medication_count_3m_0m(med_codelist):
+    return patients.with_these_medications(
+        med_codelist,
+        between=["2019-12-01", "2020-02-29"],
+        returning="number_of_matches_in_period",
+        return_expectations={
+            "int": {"distribution": "normal", "mean": 3, "stddev": 2},
+            "incidence": 0.1,
+        },
+    )
+
+def medication_count_6m_3m(med_codelist):
+    return patients.with_these_medications(
+        med_codelist,
+        between=["2019-09-01", "2020-11-30"],
+        returning="number_of_matches_in_period",
+        return_expectations={
+            "int": {"distribution": "normal", "mean": 3, "stddev": 2},
+            "incidence": 0.1,
+        },
+    )
+
+def medication_count_12m_6m(med_codelist):
+    return patients.with_these_medications(
+        med_codelist,
+        between=["2019-03-01", "2020-08-31"],
+        returning="number_of_matches_in_period",
+        return_expectations={
+            "int": {"distribution": "normal", "mean": 3, "stddev": 2},
+            "incidence": 0.1,
+        },
+    )
+
+def medication_earliest(med_codelist):
+    return patients.with_these_medications(
+        med_codelist,
+        between=["2010-01-01", "2020-02-29"],
+        returning="date",
+        find_first_match_in_period=True,
+        include_month=True,
+        return_expectations={
+            "incidence": 0.2,
+            "date": {"earliest": "2010-01-01", "latest": "2020-02-29"},
+        },
+    )
+
+def medication_latest(med_codelist):
+    return patients.with_these_medications(
+        med_codelist,
+        between=["2010-01-01", "2020-02-29"],
```

Comment:
For meds in primary care this will be true but for the secondary care high costs medicines this will only be the last two years

Reply:
We're no longer pulling the latest date since it's not used in the STATA, but I've left in the function in case it's helpful in future.